src/shell/command_line.cpp

```diff
@@ -322,49 +322,27 @@ uint16_t CommandLine::Get_arglength()
 	return static_cast<uint16_t>(total_length);
 }
 
-CommandLine::CommandLine(const char *name, const char *cmdline)
+CommandLine::CommandLine(const std::string_view name, const std::string_view cmdline)
+        : file_name(name)
 {
-	if (name) {
-		file_name = name;
-	}
-
-	// Parse the commands and put them in the list
-	bool inword, inquote;
-	char c;
-	inword  = false;
-	inquote = false;
-	std::string str;
-
-	const char* c_cmdline = cmdline;
-
-	while ((c = *c_cmdline) != 0) {
-		if (inquote) {
-			if (c != '"') {
-				str += c;
-			} else {
-				inquote = false;
-				cmds.push_back(str);
-				str.erase();
+	decltype(cmds)::value_type argument;
```

Comment:
Suggest a `std::string argument = {};` because it's accruing character-by-character from the incoming std::string_view. 

`decltype(cmds)::value_type` is an eyeful and generally we want to avoid this (if we adopted it whole-sale, it would riddle the code base).

Normally reserve it when the source type (or an interfacing type) is even uglier - or there's ongoing risk that the source type itself is changing through ongoing and frequent development or perhaps even a template type.